google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -117,26 +126,27 @@ private AppEngineUtil() {
   public static List<ModuleDeploymentSource> createModuleDeploymentSources(
       @NotNull Project project) {
     AppEngineProjectService projectService = AppEngineProjectService.getInstance();
-    AppEngineAssetProvider assetProvider = AppEngineAssetProvider.getInstance();
 
     List<ModuleDeploymentSource> moduleDeploymentSources = Lists.newArrayList();
 
     boolean hasStandardModules = false;
 
     for (Module module : ModuleManager.getInstance(project).getModules()) {
-      XmlFile appEngineWebXml = assetProvider.loadAppEngineStandardWebXml(
-          project, Collections.singletonList(module));
-
-      AppEngineEnvironment environment
-          = projectService.getModuleAppEngineEnvironment(appEngineWebXml);
-
-      if (ModuleType.is(module, JavaModuleType.getModuleType())
-          && projectService.isJarOrWarMavenBuild(module)) {
-        moduleDeploymentSources.add(createMavenBuildDeploymentSource(project, module, environment));
-      }
+      FacetManager facetManager = FacetManager.getInstance(module);
+      if (facetManager.getFacetByType(AppEngineStandardFacet.ID) != null
+          || facetManager.getFacetByType(AppEngineFlexibleFacetType.ID) != null) {
+        AppEngineEnvironment environment = projectService.getModuleAppEngineEnvironment(module)
+            .orElseThrow(() -> new RuntimeException("No environment."));
```

Comment:
As above. We can chat about this if you prefer, it might be more constructive.
In short, yes, I don't expect it to happen, and even prefer things to blow up if it does.